### PR TITLE
Use a pager for help output

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
Automatically use a pager for the help

Depends on https://github.com/minio/cli/pull/8

## Motivation and Context
Add pagination when printing long help message

## How to test this PR?
mc cp -h

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by klauspost** [comment]
@vadmeste Would you mind if I try to implement without invoking less?

**Comment by vadmeste** [comment]
> @vadmeste Would you mind if I try to implement without invoking less?

No problem, do you know any good library for that ? also I found it this PR doesn't work in some cases, unfortunately the cli library calls os.Exit() in certain cases, this is maybe not easy to implement

**Comment by klauspost** [comment]
Yeah, `os.Exit` is always going to be a problem. So this only for help (for now?)

While a full 'less' would be great, maybe just paging would be fine.

External programs are always going to be flaky, so while it has the upside of all the "less" features, if they need that then `mc --help|less` is the way to go.

**Comment by vadmeste** [comment]
> Rather than using less you can use this https://github.com/charmbracelet/bubbletea/tree/master/examples/pager @vadmeste

Yes, I didn't see this when I went through their docs, will try it out

**Comment by vadmeste** [comment]
I made some progress, the code currently is little bit complex because I had to connect cli with bubbletea but overall it is fine, can you test this @harshavardhana before I make the remaining changes ? hopefully no surprises after that

**Comment by harshavardhana** [comment]
It works fine @vadmeste however we have to show how to `quit` at the bottom page otherwise users won't know how to quit the pager.

**Comment by vadmeste** [comment]
> It works fine @vadmeste however we have to show how to quit at the bottom page otherwise users won't know how to quit the pager.

okay, there are still more edge cases that I need to address, they should be fixed today

**Comment by vadmeste** [comment]
This PR https://github.com/minio/cli/pull/9/files is also needed. Otherwise all of this won't work :)

File: cmd/term-pager.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
+package cmd
```
Comment (harshavardhana) [inline cmd/term-pager.go:1]: license header missing @vadmeste

File: main.go [hunk 2]
```diff
@@ -24,5 +24,7 @@ import (
 )
 
 func main() {
-	mc.Main(os.Args)
+	if e := mc.Main(os.Args); e != nil {
+		os.Exit(1)
```
Comment (harshavardhana) [inline main.go:28]:
```suggestion
		console.Fatalln(e)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: go.mod [hunk 3]
```diff
@@ -16,7 +16,7 @@ require (
 	github.com/klauspost/compress v1.15.9
 	github.com/mattn/go-ieproxy v0.0.1
 	github.com/mattn/go-isatty v0.0.14
-	github.com/minio/cli v1.23.0
+	github.com/minio/cli v1.23.1-0.20220921164421-38105e9ed822
```
Comment (harshavardhana) [inline go.mod:19]: tagged v1.24.0 @vadmeste

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by harshavardhana** [comment]
PTAL at the conflict @vadmeste

**Review by harshavardhana — APPROVED** [review]

**Comment by zeeZ** [comment]
Is there an option included to disable this?

It's pretty cumbersome if the help output disappears while you're trying to cobble together a command with multiple flags and one can't look at the output history because it disappears with the pager.

`mc cp -h | cat` is a dumb workaround.

**Comment by harshavardhana** [comment]
We are going to fix the pager for it to stay the output..

**Comment by vadmeste** [comment]
> Is there an option included to disable this?

@zeeZ I think you want this PR, https://github.com/minio/mc/pull/4289, it keeps the help screen after quitting it

**Pull request merged**